administrator/components/com_content/models/article.php

```diff
@@ -681,6 +679,61 @@ public function featured($pks, $value = 0)
 	}
 
 	/**
+	 * Method to store the token generated.
+	 *
+	 * @return  boolean  True on success.
+	 *
+	 * @since 3.7
+	 */
+	private function shareTokenGenerate()
+	{
+		jimport('joomla.user.helper');
+		$token = JUserHelper::genRandomPassword(16);
+
+		return $token;
+	}
+
+	/**
+	 * Method to store the token generated.
+	 *
+	 * @param   string  $title  The title of the shared article.
+	 *
+	 * @return  boolean  True on success.
+	 *
+	 * @since 3.7
+	 */
+	public function shareToken($title)
+	{
+		$table = $this->getTable('share', 'ContentTable');
+		$token = $this->shareTokenGenerate();
+
+		try
+		{
+			$db = $this->getDbo();
+			$query = $db->getQuery(true);
+
+			$data = array('articleId', 'title', 'sharetoken');
+			$values = array($db->quote($this->get('id')), $db->quote($title), $db->quote($token));
+
+			$query
+				->insert($db->quoteName('#__share_draft'))
+				->columns($db->quoteName($data))
+				->values(implode(',', $values));
+
+
+			$db->setQuery($query);
+			$table->save($data);
+
+			return $db->execute();
+		}
+		catch (Exception $e)
+		{
+			Jtext::_('COM_CONTENT_TOKEN_ERROR');
```

Comment:
You are missing an echo here (JText does not output anything). But Roland means (imo) that you just should remove try and catch completely.